### PR TITLE
Enable modules to depend on CPU architecture, Update OpenJFX and support Aarch64 for Linux and mac OS

### PR DESCRIPTION
**Pull request by matthiasblaesing**
OpenJFX JARs are not only OS, but also architecture specific. Before this change it was possible to load OpenJFX when the OS matched, but the architecture was wrong. This is fixed by adding the ability to depend on a CPU architecture to to the NetBeans module system.

Similar to the `org.openide.modules.os.<OS>` tokens, a second set of tokens is provided as `org.openide.modules.arch.<ARCH>`. `<ARCH>` matches the value of `os.arch` which is in line with the naming convention for JNI library loading in NetBeans.

The existing three OpenJFX modules were modified to require `org.openide.modules.arch.amd64` (Linux and Windows) or `org.openide.modules.arch.x86_64` (mac OS).

As aarch64 gains traction the existing OpenJFX module set was completed by variants for Linux and mac OS Aarch64.

Closes: #5985

### CODE REVIEW

**Review by lkishalmi — APPROVED** [review]
Nice work!

File: platform/core.startup/src/org/netbeans/core/startup/ConsistencyVerifier.java [hunk 1]
```diff
@@ -91,7 +99,20 @@ public static SortedMap<String,SortedSet<String>> findInconsistencies(
...
+                "org.openide.modules.arch.x86," + // NOI18N
+                "org.openide.modules.arch.aarch64," + // NOI18N
+                "org.openide.modules.arch.arm," + // NOI18N
+                "org.openide.modules.arch.ppc," + // NOI18N
+                "org.openide.modules.arch.ppc64," + // NOI18N
+                "org.openide.modules.arch.ppc64le," + // NOI18N
+                "org.openide.modules.arch.s390x," + // NOI18N
+                "org.openide.modules.arch.zarch_64," + // NOI18N
+                "org.openide.modules.arch.sparc" + // NOI18N
+                "org.openide.modules.arch.sparcv9" + // NOI18N
```
[the remark this answers is not in the document]
Comment (matthiasblaesing) [reply platform/core.startup/src/org/netbeans/core/startup/ConsistencyVerifier.java]: Valid point. Pushed update.

**Review by mbien — APPROVED** [review]
looks good!

**Comment by mbien** [comment]
restarting with platform tests just to be sure.

**Comment by mbien** [comment]
ActivatorTest needs an update:
```
junit.framework.AssertionFailedError: expected:
<[cnb.org.openide.modules, org.openide.modules.os.Windows]> but was:
<[cnb.org.openide.modules, org.openide.modules.arch.amd64, org.openide.modules.os.Windows]>
  at org.netbeans.core.osgi.ActivatorTest.testProvidesRequiresNeedsParsing(ActivatorTest.java:180)
  at org.netbeans.junit.NbTestCase.access$200(NbTestCase.java:79)
  at org.netbeans.junit.NbTestCase$2.doSomething(NbTestCase.java:483)
  at org.netbeans.junit.NbTestCase$1Guard.run(NbTestCase.java:404)
  at java.lang.Thread.run(Thread.java:750)
```

**Comment by matthiasblaesing** [comment]
I'll let this stay open for the next few days and intent to merge sometime next week, so if anyone wants to object, no would be a good time.

File: platform/core.startup/src/org/netbeans/core/startup/ConsistencyVerifier.java [hunk 2]
```diff
@@ -91,7 +99,21 @@ public static SortedMap<String,SortedSet<String>> findInconsistencies(
...
+                "org.openide.modules.arch.amd64, " + // NOI18N
+                "org.openide.modules.arch.x86," + // NOI18N
+                "org.openide.modules.arch.aarch64," + // NOI18N
+                "org.openide.modules.arch.arm," + // NOI18N
+                "org.openide.modules.arch.ppc," + // NOI18N
+                "org.openide.modules.arch.ppc64," + // NOI18N
+                "org.openide.modules.arch.ppc64le," + // NOI18N
+                "org.openide.modules.arch.s390x," + // NOI18N
+                "org.openide.modules.arch.zarch_64," + // NOI18N
+                "org.openide.modules.arch.sparc," + // NOI18N
```
Comment (BradWalker) [inline platform/core.startup/src/org/netbeans/core/startup/ConsistencyVerifier.java:113]: Hey @matthiasblaesing , I really don't like that we still support SPARC. It's a long time dead platform.

Also, not a fan of including SPARC64 support. As it's been removed from the JDK for a little while.

**Review by jtulach — APPROVED** [review]

**Pull request merged**